app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -1315,8 +1309,38 @@ else if (CommCareApplication._().getAppResourceState() != CommCareApplication.ST
                 //Normal Home Screen login time!
                 refreshView();
             }
-        } catch (SessionUnavailableException sue) {
-            //TODO: See how much context we have, and go login
+        } catch(SessionUnavailableException sue) {
+            returnToInitialScreen();
+        }
+    }
+
+    /**
+     * Called in the event that we are leaving the CCHomeActivity (generally due to there no
+     * longer being an active session). Makes the decision of which prior screen or application
+     * state to return to, depending on the state of all currently installed CC apps.
+     */
+    private void returnToInitialScreen() {
+        // 1) If there is exactly one visible app and it's missing its MM, redirect to MM
+        // verification (because we're assuming the user is not using multiple apps)
+        if (CommCareApplication._().getVisibleAppRecords().size() == 1 && 
```

Comment:
The coupling to me between "Visible"/"Ready" and "Missing Multimedia" isn't really super apparent.

In my mind the process here seems like it should be something like:

dispatchHomeScreen()
   if !seatApp() {
       initApp()
   }
   returnToInitial

returnToInitialScreen()

  if no seated app:
    seatApp()
  if currentapp.needsValidation()
     run commcare verification activity
  else
    return to login

seatapp():
   if no app records:
    return false
  if readyRecords = 1:
    seat
 if pending = 1
    seat
 else
    go to app manager (or trigger app exit)